### ISSUE TITLE
Mousing over button text shouldn't show cursor=text

### ISSUE THREAD

**Issue by frewsxcv** [servo/servo#9084]
Comparing Servo and Chrome on [this page](https://news.ycombinator.com/item?id=10809486):

![untitled](https://cloud.githubusercontent.com/assets/416575/12044142/818bea68-ae41-11e5-8ed3-b5eb3f2ce76a.gif)

**Comment by jdm** [comment]
This comes from `mouse_over` in `components/layout/query.rs`, but originates from the code in `display_list_builder.rs` which stores cursors for text fragments in `build_display_list_for_text_fragment` and `build_display_items_for_selection_if_necessary`. I guess we either need a way of overriding that or explicit checks to see if the text fragment is in a button. I guess we could try adding a `cursor` property to the `button` rules in the UA stylesheet, though?

**Comment by frewsxcv** [comment]
This might be the associated line in Gecko?

https://github.com/mozilla/gecko-dev/blob/e192050ec9987883ea76c85950301dd5c4314a0c/layout/style/forms.css#L628

**Comment by lumiscosity** [comment]
Still happens. For future reference: this button is an `<input type="submit" value="add comment">`. A regular `<button>` works fine. Tested on Arch Linux x86_64, Servo built against commit 633d746e2f1f0bb397e884381bbb7fab89284006, profiling build.

<img width="545" height="639" alt="Image" src="https://github.com/user-attachments/assets/b117fc4f-1e08-487d-9e75-af956b50888c" />

**Comment by jdm** [comment]
Interesting: https://github.com/servo/servo/blob/989c0d8994da9783f8ced75b2bf2399b7a12869c/components/layout/stylesheets/servo.css#L153-L154

```html
<input type="submit" value="submit">
<input type="reset" value="reset">
<input type="button" value="button">
<input type="checkbox" checked>
<input type="radio" checked>
```

I think it's not enough to use :not selectors in that stylesheet, because we still end up setting the cursor based on the text that's under it. Instead we need something like:
```css
input:is([type=submit i], [type=reset i], /*etc*/) {
  cursor: default;
}
```

**Comment by jdm** [comment]
I don't think we have a good way to write an automated test for this, so manual testing should be fine.
